charms/worker/k8s/src/token_distributor.py

```diff
@@ -305,13 +344,20 @@ def revoke_tokens(
 
         status.add(ops.MaintenanceStatus(f"Revoking {token_type.value} tokens"))
         for unit in remove:
-            if hostname := app_databag.get(unit.name):
+            if node_state := app_databag.get(unit.name):
+                state, node = node_state.split("-", 1)
                 log.info(
-                    "Revoking token for %s unit=%s:%s hostname=%s",
+                    "Revoking token for %s unit=%s:%s %s node=%s",
                     token_type.value,
                     relation.name,
                     unit.name,
-                    hostname,
+                    state,
+                    node,
                 )
-                token_strat(hostname, token_type)
+                # ignore pending state, because we can't confirm
+                # if the other node ever used the token to join
+                # ignore self revocation
+                ignore_errors = self.node_name == node or state == "pending"
+                token_strat(node, ignore_errors)
```

Comment:
We can `ignore_errors` on the pending unit.  Here's why -- maybe we gave a token to that unit that it never used b/c it failed to get there.  SO we could hit failures on removing a node that's not in the cluster.  Since there's NO API in the snap to tell me which nodes are in the cluster (???? I mean really ????) We have to keep up with POSSIBLE nodes in the `app_databag`